packages/remix/src/build.ts

```diff
@@ -0,0 +1,266 @@
+    } else if (hasScript('build', pkg)) {
+      debug(`Executing "yarn build"`);
+      await runPackageJsonScript(entrypointFsDirname, 'build', spawnOpts);
+    } else {
+      await execCommand('remix build', {
+        ...spawnOpts,
+        cwd: entrypointFsDirname,
+      });
+    }
+  }
+
+  let serverBuildPath = 'build/index.js';
+  let needsHandler = true;
+  try {
+    const remixConfig: AppConfig = require(join(
+      entrypointFsDirname,
+      'remix.config'
+    ));
+
+    // If `serverBuildTarget === 'vercel'` then Remix will output a handler
+    // that is already in Vercel (req, res) format, so don't inject the handler
+    if (remixConfig.serverBuildTarget) {
+      if (remixConfig.serverBuildTarget !== 'vercel') {
+        throw new Error(
+          `\`serverBuildTarget\` in Remix config must be "vercel" (got "${remixConfig.serverBuildTarget}")`
+        );
+      }
+      serverBuildPath = 'api/index.js';
+      needsHandler = false;
+    }
+
+    if (remixConfig.serverBuildPath) {
+      // Explicit file path where the server output file will be
+      serverBuildPath = remixConfig.serverBuildPath;
+    } else if (remixConfig.serverBuildDirectory) {
+      // Explicit directory path the server output will be
+      serverBuildPath = join(remixConfig.serverBuildDirectory, 'index.js');
+    }
+  } catch (err: any) {
+    // Ignore error if `remix.config.js` does not exist
+    if (err.code !== 'MODULE_NOT_FOUND') throw err;
+  }
+
+  const [staticFiles, renderFunction] = await Promise.all([
+    glob('**', join(entrypointFsDirname, 'public')),
+    createRenderFunction(
+      entrypointFsDirname,
+      serverBuildPath,
+      needsHandler,
+      nodeVersion
+    ),
+  ]);
+
+  return {
+    routes: [
+      {
+        src: '^/build/(.*)$',
+        headers: { 'cache-control': 'public, max-age=31536000, immutable' },
+        continue: true,
+      },
+      {
+        handle: 'filesystem',
+      },
+      {
+        src: '/(.*)',
+        dest: '/render',
+      },
+    ],
+    output: {
+      render: renderFunction,
+      ...staticFiles,
+    },
+  };
+};
+
+function hasScript(scriptName: string, pkg: PackageJson | null) {
+  const scripts = (pkg && pkg.scripts) || {};
+  return typeof scripts[scriptName] === 'string';
+}
+
+async function createRenderFunction(
+  rootDir: string,
+  serverBuildPath: string,
+  needsHandler: boolean,
+  nodeVersion: NodeVersion
+): Promise<NodejsLambda> {
+  const files: Files = {};
+  const handler = needsHandler
+    ? join(dirname(serverBuildPath), '__vc_handler.js')
+    : serverBuildPath;
+  const handlerPath = join(rootDir, handler);
+
+  if (needsHandler) {
+    // Copy the `default-server.js` file into the "build" directory
+    const sourceHandlerPath = join(__dirname, '../default-server.js');
+    await fs.copyFile(sourceHandlerPath, handlerPath);
+  }
+
+  // Trace the handler with `@vercel/nft`
+  const trace = await nodeFileTrace([handlerPath], {
+    base: rootDir,
```

Comment:
I think the base won't work if this is a monorepo because it should be repoRoot

Reply:
Was this missed?

I've been unable to use the Remix preset since Friday because it's not finding the yarn.lock file in the root of my monorepo and I think this might be the cause.

Reply:
Please try this https://github.com/vercel/vercel/pull/8077#pullrequestreview-1033551949